state-chain/pallets/cf-ingress-egress/src/boost_pool/tests.rs

```diff
@@ -160,19 +162,82 @@ fn boosting_with_fees() {
 
 	check_pool(&pool, [(BOOSTER_1, 1000), (BOOSTER_2, 2000)]);
 
-	assert_eq!(pool.provide_funds_for_boosting(BOOST_1, 1010), Ok((1010, 10)));
+	assert_eq!(pool.provide_funds_for_boosting(BOOST_1, 1010, NO_DEDUCTION), Ok((1010, 10)));
 
 	// The recorded amounts include fees (1 is missing due to rounding errors in *test* code)
 	check_pending_boosts(
 		&pool,
 		[(BOOST_1, vec![(BOOSTER_1, 333 + 3, 3), (BOOSTER_2, 667 + 6, 6)])],
 	);
 
-	assert_eq!(pool.process_deposit_as_finalised(BOOST_1), vec![]);
+	assert_eq!(
+		pool.process_deposit_as_finalised(BOOST_1),
+		DepositFinalisationOutcomeForPool { amount_credited: 1010, unlocked_funds: vec![] }
+	);
 
 	check_pool(&pool, [(BOOSTER_1, 1003), (BOOSTER_2, 2006)]);
 }
 
+#[test]
+fn boosting_with_fees_including_network_fee_portion() {
+	const NETWORK_FEE_PORTION_PERCENT: u8 = 30;
```

Comment:
Worth testing the extreme case of 100% too, just to see it behaves rationally.
(0 is tested in some of the other tests already).